src/domain/usecases/cache/get-cache-value.ts

```diff
@@ -0,0 +1,16 @@
+export interface GetCacheValue {
+  get(params: GetCacheValue.Params): GetCacheValue.Result;
+}
+export namespace GetCacheValue {
+  export type Params = {
+    key: string;
+    throws?: boolean;
+    options: { parseToJson: boolean; parseBufferToString: boolean };
```

Comment:
Create an enum with the format types.

`type AllowedTypes = 'JSON' | 'STRING'`

If this property is set, you perform the conversion.